src/jvm/main/org/jetbrains/kotlinx/lincheck/TransformationClassLoader.java

```diff
@@ -49,18 +49,12 @@ public class TransformationClassLoader extends ExecutionClassLoader {
     private static final Map<String, byte[]> stressStrategyBytecodeCache = new ConcurrentHashMap<>();
     private static final Map<String, byte[]> modelCheckingStrategyBytecodeCache = new ConcurrentHashMap<>();
 
-    public TransformationClassLoader(Strategy strategy, Runner runner) {
+    public TransformationClassLoader(Strategy strategy) {
         classTransformers = new ArrayList<>();
         // Apply the strategy's transformer at first, then the runner's one.
-        if (strategy.needsTransformation()) classTransformers.add(strategy::createTransformer);
-        if (runner.needsTransformation()) classTransformers.add(runner::createTransformer);
-        remapper = UtilsKt.getRemapperByTransformers(
-                // create transformers just for their class information
-                classTransformers.stream()
-                        // pass any parameter to Transformer constructors - it won't be used
-                        .map(constructor -> constructor.apply(new TraceClassVisitor(null)))
-                        .collect(Collectors.toList())
-        );
+        TransformationMode transformationMode = strategy instanceof ModelCheckingStrategy ? TransformationMode.MODEL_CHECKING : TransformationMode.STRESS;
```

Comment:
It would be better to have a `Strategy.transformationMode` property instead.

Reply:
In this case, we have to expose the `TransformationMode` enum as public (now it's internal) because otherwise a call it from Java code will look like this:
<img width="600" alt="image" src="https://github.com/JetBrains/lincheck/assets/45040429/521b8378-f798-497e-9963-76be9e079aab">

Reply:
So, I suggest postponing this refactoring until TransformationClassLoader is present and is written in Java.

Reply:
Ok, please add a TODO then